apps/balancer-tools/src/app/poolsimulator/(components)/SwapSimulator.tsx

```diff
@@ -290,3 +143,293 @@ function calculateSimulation({
     priceImpact,
   };
 }
+
+function SwapSimulatorForm({
+  setInitialResult,
+  setCustomResult,
+}: {
+  setInitialResult: React.Dispatch<React.SetStateAction<IResult>>;
+  setCustomResult: React.Dispatch<React.SetStateAction<IResult>>;
+}) {
+  const { initialData, initialAMM, customAMM } = usePoolSimulator();
+
+  if (!initialData || !initialAMM || !customAMM) return <Spinner />;
```

Comment:
If you do this you'll be conditionally calling `useEffect` (below), which breaks one of the [Rules of Hooks](https://react.dev/warnings/invalid-hook-call-warning)